src/turbomind/models/llama/flash_attention2/flash_fwd_launch_template.h

```diff
@@ -14,7 +14,13 @@
 template<typename Kernel_traits, bool Is_dropout, bool Is_causal, bool Is_even_N, bool Is_even_K, bool Return_softmax>
 __global__ void flash_fwd_kernel(Flash_fwd_params params)
 {
+
+#if __CUDA_ARCH__ >= 800
     flash::compute_attn<Kernel_traits, Is_dropout, Is_causal, Is_even_N, Is_even_K, Return_softmax>(params);
+#else
+// TODO: support flash attention2 on sm<80
+    assert(false);
```

Comment:
Will it affect w4a16 inference?

Reply:
https://github.com/InternLM/lmdeploy/blob/c07f60fd0abb76e6d6a066636b408630422a9f1f/src/turbomind/models/llama/llama_kernels.cu#L800

device with sm<80 would fallback to flash attention2. This branch should never been reached at runtime.